test/Microsoft.VisualStudio.Editor.Razor.Test/BraceSmartIndenterTest.cs

```diff
@@ -0,0 +1,382 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Editor.Razor.Test.Infrastructure;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public class BraceSmartIndenterTest : ForegroundDispatcherTestBase
+    {
+        [Fact]
+        public void TextBuffer_OnPostChanged_MovesToEndOfLine()
+        {
+            // Arrange
+            var change = Environment.NewLine;
```

Comment:
This isn't right, we need to work with all newline types in all places, we shouldn't test based on `Environment.NewLine`

Reply:
Ya, getting all the test infrastructure to work with all variants of newline is painful which is why I stuck to `Environment.NewLine`. The product code on the other hand does understand all aspects. I'll put in some time to make the test infrastructure more resilient.